packages/salesforcedx-vscode-core/src/commands/isvdebugging/bootstrapCmd.ts

```diff
@@ -512,30 +511,55 @@ export type IsvDebugBootstrapConfig = ProjectNameAndPath & ForceIdeUri;
 export interface ForceIdeUri {
   loginUrl: string;
   sessionId: string;
+  orgName: string;
 }
 
 export class EnterForceIdeUri implements ParametersGatherer<ForceIdeUri> {
+  public forceIdUrl?: ForceIdeUri;
   public async gather(): Promise<
     CancelResponse | ContinueResponse<ForceIdeUri>
   > {
     const forceIdeUri = await vscode.window.showInputBox({
-      prompt: nls.localize('parameter_gatherer_paste_forceide_url')
+      prompt: nls.localize('parameter_gatherer_paste_forceide_url'),
+      placeHolder: nls.localize(
+        'parameter_gatherer_paste_forceide_url_placeholder'
+      ),
+      ignoreFocusOut: true,
+      validateInput: value => {
+        try {
+          const url = new URL(value);
+          const parameter = url.searchParams;
+          const loginUrl = parameter.get('url');
+          const sessionId = parameter.get('sessionId');
+          if (typeof loginUrl !== 'string' || typeof sessionId !== 'string') {
+            return nls.localize('parameter_gatherer_invalid_forceide_url');
+          }
+        } catch (e) {
+          return nls.localize('parameter_gatherer_invalid_forceide_url');
+        }
+
+        return null; // all good
+      }
     });
 
     if (forceIdeUri) {
-      const url = Uri.parse(forceIdeUri);
-      const parameter = querystring.parse(url.query);
-      if (parameter.url && parameter.sessionId) {
+      const url = new URL(forceIdeUri);
+      const parameter = url.searchParams;
+      const loginUrl = parameter.get('url');
+      const sessionId = parameter.get('sessionId');
+      if (loginUrl && sessionId) {
         const protocolPrefix =
-          parameter.secure === '0' ? 'http://' : 'https://';
+          parameter.get('secure') === '0' ? 'http://' : 'https://';
+        this.forceIdUrl = {
+          loginUrl: loginUrl.toLowerCase().startsWith('http')
+            ? loginUrl
+            : protocolPrefix + loginUrl,
+          sessionId: sessionId,
+          orgName: url.hostname
```

Comment:
I think this can have dashes, which should be okay on all operating systems. Can this have any other special characters, like a colon (local appserver with port)?